File: backend/app/api/schemas/agendamentos.py
"""Schemas dos agendamentos (S4)."""
from __future__ import annotations

from typing import Any

from pydantic import BaseModel


class JobOut(BaseModel):
    id: str
    nome: str
    descricao: str
    hora: int
    ligado: bool
    hora_key: str
    enabled_key: str


class JobRunResult(BaseModel):
    job: str
    executado_em: str
    resultado: dict[str, Any]

File: frontend/lib/types/agendamentos.ts
// Agendamentos (S4) — jobs do scheduler vistos na UI.

export interface Job {
  id: string;
  nome: string;
  descricao: string;
  hora: number;
  ligado: boolean;
  hora_key: string; // chave de config p/ editar a hora (via S3)
  enabled_key: string; // chave de config p/ ligar/desligar (via S3)
}

export interface JobRunResult {
  job: string;
  executado_em: string;
  resultado: Record<string, unknown>;
}
